File: normal buyers/find_missing_subs.py
import csv
import re
from collections import Counter

# 1. Read the deployed index.html to find existing subdivisions
index_path = r"c:\Users\daved\AntiGravity Projects\follow up boss\normal buyers\deploy\index.html"
with open(index_path, 'r', encoding='utf-8') as f:
    content = f.read()

# Try to extract names. Let's look for name: "..." or title: "..."
existing_names = set(re.findall(r'name:\s*["\']([^"\']+)["\']', content))
if not existing_names:
    existing_names = set(re.findall(r'title:\s*["\']([^"\']+)["\']', content))

print(f"Found {len(existing_names)} existing subdivisions in index.html")
# print("Existing:", existing_names)

# 2. Read the full CSV to find ALL subdivisions and their counts
csv_path = r"C:\Users\daved\AntiGravity Projects\follow up boss\normal buyers\Default_MLS_Defined_Spreadsheet (1).csv"

subdivisions = []
with open(csv_path, 'r', encoding='utf-8') as f:
    reader = csv.DictReader(f)
    for row in reader:
        sub = row.get('Subdivision', '').strip()
        city = row.get('City', '').strip()
        # We only want Gulf Shores
        if sub and city.lower() == 'gulf shores':
            subdivisions.append(sub)

counts = Counter(subdivisions)

# 3. Filter out the ones we already have
missing = []
for sub, count in counts.most_common():
    # Basic string matching, lowercasing to be safe
    # We might have slight variations like "Aventura" vs "Aventura Phase 1"
    is_mapped = False
    for existing in existing_names:
        if sub.lower() in existing.lower() or existing.lower() in sub.lower():
            is_mapped = True
            break
    
    if not is_mapped:
        missing.append((sub, count))

print("\nTop Missing Subdivisions in Gulf Shores (by number of listings):")
for sub, count in missing[:30]:
    print(f"- {sub} ({count} listings)")

File: normal buyers/find_missing_subs.js
const fs = require('fs');

const index_path = "c:\\Users\\daved\\AntiGravity Projects\\follow up boss\\normal buyers\\deploy\\index.html";
const csv_path = "C:\\Users\\daved\\AntiGravity Projects\\follow up boss\\normal buyers\\Default_MLS_Defined_Spreadsheet (1).csv";

const content = fs.readFileSync(index_path, 'utf8');
const nameRegex = /name:\s*["']([^"']+)["']/g;

const existingNames = new Set();
let match;
while ((match = nameRegex.exec(content)) !== null) {
    if (match[1].trim() !== '') {
        existingNames.add(match[1].toLowerCase().trim());
    }
}
console.log("Existing names mapping: ", existingNames);

const csvContent = fs.readFileSync(csv_path, 'utf8');
const rows = csvContent.split('\n');
const headers = rows[0].split(',');
const subIndex = headers.findIndex(h => h.includes('Subdivsn'));
const cityIndex = headers.findIndex(h => h.includes('City'));

if (subIndex === -1 || cityIndex === -1) {
    console.log("Couldn't find columns");
    process.exit(1);
}

const counts = {};

for (let i = 1; i < rows.length; i++) {
    const row = rows[i].split(',');
    if (row.length > Math.max(subIndex, cityIndex)) {
        const sub = row[subIndex].replace(/["']/g, '').trim();
        const city = row[cityIndex].replace(/["']/g, '').trim();
        
        if (city.toLowerCase() === 'gulf shores' && sub) {
            counts[sub] = (counts[sub] || 0) + 1;
        }
    }
}

const missing = [];
for (const [sub, count] of Object.entries(counts)) {
    let isMapped = false;
    for (const existing of existingNames) {
        if (existing.length > 3 && (sub.toLowerCase().includes(existing) || existing.includes(sub.toLowerCase()))) {
            isMapped = true;
            break;
        }
    }
    if (!isMapped && sub !== "None" && sub !== "Not in a Subdivision" && !sub.includes("Oyster Bay Village")) {
        missing.push({ sub, count });
    }
}

missing.sort((a, b) => b.count - a.count);

console.log("\nTop Missing Subdivisions in Gulf Shores:");
for (let i = 0; i < Math.min(30, missing.length); i++) {
    console.log(`- ${missing[i].sub} (${missing[i].count} listings)`);
}
